test/e2e/node_feature_discovery.go

```diff
@@ -100,6 +101,7 @@ func cleanupNode(cs clientset.Interface) {
 // Actual test suite
 var _ = SIGDescribe("Node Feature Discovery", func() {
 	f := framework.NewDefaultFramework("node-feature-discovery")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
```

Comment:
Here, for nfd-master and nfd-worker we shouldn't need this, right?

Reply:
Here we have a choice, but we can't continue with the existing configuration since the test is falling right at the beginning. The choices are:
1. Adding to the pod spec all the necessary security constraints such as:
 `RunAsNonRoot: true`, `AllowPrivilegeEscalation: false`, etc.
2. Edit the namespace as we have here. It's the simplest solution and since this is only for testing purposes I assumed it's good enough.

Anyhow, we can go with option number 1 as well. Please let me know what is your preference.

Reply:
> 1. Adding to the pod spec all the necessary security constraints such as:
>    `RunAsNonRoot: true`, `AllowPrivilegeEscalation: false`, etc.

Ach, these are missing in the e2e pod spec 🤦 I think we definitely want to add them and this would be the preferable solution. This way the e2e will also more closely resemble the actual (default) deployment.

Reply:
Done!

Please consider that some tests are still falling but are in the `node_feature_discovery.go` division (which I am unfortunately not too experienced with). 
Issue: https://github.com/kubernetes-sigs/node-feature-discovery/issues/962

Reply:
But IMVHO I don't think it should stop us from moving on with this PR